framework/mesh/mesh_mapping/mesh_mapping.cc

```diff
@@ -0,0 +1,160 @@
+// SPDX-FileCopyrightText: 2024 The OpenSn Authors <https://open-sn.github.io/opensn/>
+// SPDX-License-Identifier: MIT
+
+#include "framework/mesh/mesh_mapping/mesh_mapping.h"
+#include "framework/logging/log.h"
+#include "framework/mesh/cell/cell.h"
+#include "framework/mesh/mesh_continuum/mesh_continuum.h"
+#include "framework/math/spatial_discretization/finite_element/piecewise_linear/piecewise_linear_continuous.h"
+
+#include <sstream>
+
+namespace opensn
+{
+
+const std::size_t MeshMapping::invalid_face_index = std::numeric_limits<std::size_t>::max();
+
+MeshMapping::CoarseMapping::CoarseMapping(const Cell& coarse_cell)
+  : fine_faces{coarse_cell.faces.size()}
+{
+}
+
+MeshMapping::FineMapping::FineMapping(const Cell& fine_cell)
+  : coarse_cell(nullptr), coarse_faces(fine_cell.faces.size(), MeshMapping::invalid_face_index)
+{
+}
+
+void
+MeshMapping::Build(const MeshContinuum& fine_grid, const MeshContinuum& coarse_grid)
+
+{
+  if (opensn::mpi_comm.size() > 1)
+    OpenSnLogicalError("MeshMapping is not currently supported in parallel.");
+  if (fine_grid.GetDimension() != coarse_grid.GetDimension())
+    OpenSnLogicalError("Grid dimensions are not equal for mapping. Fine dimension = " +
+                       std::to_string(fine_grid.GetDimension()) +
+                       ", coarse dimension = " + std::to_string(coarse_grid.GetDimension()) + ".");
+
+  coarse_to_fine_.clear();
+  fine_to_coarse_.clear();
+
+  // Instantiate the maps; *Mapping constructor takes the cell to size the face maps.
```

Comment:
Is the star (`*`) in front of "Mapping" a typo?

Reply:
Nope. Supposed to lazily represent [Coarse,Fine]Mapping

Reply:
Fairly common syntax to use for such a case is  also `{Coarse|Fine}Mapping`. But your edit is good too...